src/textbracket.ts

```diff
@@ -11,29 +11,63 @@ import { Vex } from './vex';
 import { Flow } from './tables';
 import { Element } from './element';
 import { Renderer } from './renderer';
+import { FontInfo, RenderContext } from './types/common';
+import { Note } from './note';
+
+export interface TextBracketParams {
+  start: Note;
+  stop: Note;
+  text: string;
+  superscript: string;
+  position: number | string;
+}
 
 // To enable logging for this class. Set `Vex.Flow.TextBracket.DEBUG` to `true`.
-function L(...args) {
+function L(
+  // eslint-disable-next-line
+  ...args: any[]) {
   if (TextBracket.DEBUG) Vex.L('Vex.Flow.TextBracket', args);
 }
 
+export enum TextBracketPosition {
+  TOP = 1,
+  BOTTOM = -1,
+}
+
 export class TextBracket extends Element {
-  // FIXME: Modifier.Position is singular while this is plural, make consistent
-  static get Positions() {
-    return {
-      TOP: 1,
-      BOTTOM: -1,
-    };
+  static DEBUG: boolean;
+
+  render_options: {
+    dashed: boolean;
+    color: string;
+    line_width: number;
+    underline_superscript: boolean;
+    show_bracket: boolean;
+    dash: number[];
+    bracket_height: number;
+  };
+
+  protected readonly text: string;
+  protected readonly superscript: string;
+  protected readonly position: TextBracketPosition;
+
+  protected line: number;
+  protected start: Note;
+  protected stop: Note;
+  protected font: FontInfo;
+
+  static get Position(): typeof TextBracketPosition {
```

Comment:
This changes the interface -- can you add deprecated methods that log warnings `static get Positions()` and `static get PositionsString()`.

Reply:
done